src/protocol/LSP.Synchronization.pas

```diff
@@ -401,6 +401,9 @@ procedure TDidChangeTextDocument.Process(var Params : TDidChangeTextDocumentPara
         else }
         Code.Source := TTextDocumentContentChangeEvent(Change).text;
 
+        // Ryan, uncomment this to have a syntax check at
+        // CheckSyntax(Self.Transport,Code);
```

Comment:
If this doesn't update as you type when does it update? Usually in most servers the diagnostics are updated as you type.